kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -42,18 +64,59 @@ public VirtualKafkaClusterReconciler(Clock clock) {
 
     @Override
     public UpdateControl<VirtualKafkaCluster> reconcile(VirtualKafkaCluster cluster, Context<VirtualKafkaCluster> context) {
-        var proxyOpt = context.getSecondaryResource(KafkaProxy.class, PROXY_EVENT_SOURCE_NAME);
-        LOGGER.debug("spec.proxyRef.name resolves to: {}", proxyOpt);
+        var existingProxies = context.getSecondaryResource(KafkaProxy.class, PROXY_EVENT_SOURCE_NAME).stream()
```

Comment:
Is there a reason we have some sources named and others not? Should we be moving everything to named sources?

Reply:
The difference is the cardinality of the reference (1:1 uses the name). But it's not really doing us many favours, so I might change them all to set `Set getSecondaryResource()` method. Then we wouldn't care about names at all.